### PR TITLE
[MM-32281] SqlRoleStore/GetByName: add context to allow read from master

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Allow `SqlRoleStore.GetByName` to query master to avoid replica lag. This is an indirect read after write because we create scheme and assigned roles for the scheme at the same time. Right after we try to read the roles from the db (ie. from replica and this creates the replica lag issue).

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-32281

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: store/sqlstore/scheme_store.go [hunk 1]
```diff
@@ -269,7 +269,7 @@ func (s *SqlSchemeStore) GetByName(schemeName string) (*model.Scheme, error) {
 func (s *SqlSchemeStore) Delete(schemeId string) (*model.Scheme, error) {
 	// Get the scheme
 	var scheme model.Scheme
-	if err := s.GetReplica().SelectOne(&scheme, "SELECT * from Schemes WHERE Id = :Id", map[string]interface{}{"Id": schemeId}); err != nil {
+	if err := s.GetMaster().SelectOne(&scheme, "SELECT * from Schemes WHERE Id = :Id", map[string]interface{}{"Id": schemeId}); err != nil {
```
Comment (agnivade) [inline store/sqlstore/scheme_store.go:272]: Hmm .. I'd prefer to do this and the other queries in a transaction to make it more robust. Not needed for this PR though.

**Review by agnivade — APPROVED** [review]
That must have been torture to add all those `context.Background()` :( Thanks for doing it!

**Review by mkraft — APPROVED** [review]

**Review by furqanmlk — APPROVED** [review]
LGTM

image

image

**Pull request merged**